code/controllers/subsystem/ticker.dm

```diff
@@ -373,6 +398,208 @@ SUBSYSTEM_DEF(ticker)
+				total_antagonists[temprole] += ", [Mind.name]([Mind.key])"
+			else
+				total_antagonists.Add(temprole) //If the role doesnt exist in the list, create it and add the mob
+				total_antagonists[temprole] += ": [Mind.name]([Mind.key])"
+
+	CHECK_TICK
+
+	//Now print them all into the log!
+	log_game("Antagonists at round end were...")
+	for(var/i in total_antagonists)
+		log_game("[i]s[total_antagonists[i]].")
+
+	CHECK_TICK
+
+//Haven START - Ports miscreants from citadel, credits to Deathride58
+	for(var/datum/mind/crewMind in minds)
+		if(!crewMind.current || !crewMind.objectives.len)
+			continue
+		for(var/datum/objective/miscreant/MO in crewMind.objectives)
+			miscreants += "<B>[crewMind.current.real_name]</B> (Played by: <B>[crewMind.key]</B>)<BR><B>Objective</B>: [MO.explanation_text] <font color='grey'>(Optional)</font>"
+		for(var/datum/objective/crew/CO in crewMind.objectives)
+			if(CO.check_completion())
+				to_chat(crewMind.current, "<br><B>Your optional objective</B>: [CO.explanation_text] <font color='green'><B>Success!</B></font>")
+				successfulCrew += "<B>[crewMind.current.real_name]</B> (Played by: <B>[crewMind.key]</B>)<BR><B>Objective</B>: [CO.explanation_text] <font color='green'><B>Success!</B></font> <font color='grey'>(Optional)</font>"
+			else
+				to_chat(crewMind.current, "<br><B>Your optional objective</B>: [CO.explanation_text] <font color='red'><B>Failed.</B></font>")
+
+	if (successfulCrew.len)
+		var/completedObjectives = "<B>The following crew members completed their Crew Objectives:</B><BR>"
+		for(var/i in successfulCrew)
+			completedObjectives += "[i]<BR>"
+		to_chat(world, "[completedObjectives]<BR>")
+	else
+		if(CONFIG_GET(flag/allow_crew_objectives))
+			to_chat(world, "<B>Nobody completed their Crew Objectives!</B><BR>")
+
+	CHECK_TICK
+
+	if (miscreants.len)
+		var/miscreantObjectives = "<B>The following crew members were miscreants:</B><BR>"
+		for(var/i in miscreants)
+			miscreantObjectives += "[i]<BR>"
+		to_chat(world, "[miscreantObjectives]<BR>")
+
+	CHECK_TICK
+//Haven END - Ports miscreants from citadel, credits to Deathride58
+
+	mode.declare_station_goal_completion()
+
+	CHECK_TICK
+	//medals, placed far down so that people can actually see the commendations.
+	if(GLOB.commendations.len)
+		to_chat(world, "<b><font size=3>Medal Commendations:</font></b>")
+		for (var/com in GLOB.commendations)
+			to_chat(world, com)
+
+	CHECK_TICK
+
+	//Collects persistence features
+	if(mode.allow_persistence_save)
+		SSpersistence.CollectData()
+
+	//stop collecting feedback during grifftime
+	SSblackbox.Seal()
+
+	sleep(50)
+	ready_for_reboot = TRUE
+	standard_reboot()
+
+/datum/controller/subsystem/ticker/proc/standard_reboot()
+	if(ready_for_reboot)
+		if(mode.station_was_nuked)
+			Reboot("Station destroyed by Nuclear Device.", "nuke")
+		else
+			Reboot("Round ended.", "proper completion")
+	else
+		CRASH("Attempted standard reboot without ticker roundend completion")
```

Comment:
What in the actual fuck happened here? Why did these 200 lines get readded in the merge, when there's only 30 lines that are actually required for the round end crew objective/miscreant display to function? Port the portion inside the //haven bits to TG's new roundend message box thing, instead of trying to readd TG's old chatbox display for roundend stats.

Reply:
I'll be totally honest. No clue. Conflict fixer is pretty fuckin clunky.

Reply:
In a downstream repo like this, you should almost always prioritize upstream's changes over downstream's. Prioritizing downstream changes when fixing merge conflicts can and will result in major regressions like this one. Changes in the PR will always be at the top, and changes upstream will always be at the bottom.

Reply:
Yah I'm aware of the need to prioritize upstream shit. The issue is I can't really reliably 'do' that for this specific PR with github's built in editors. Tori has ownership of their branch and by proxy I can't directly edit shit via dreammaker far as I can tell.